types/components/buttons.d.ts

```diff
@@ -0,0 +1,6 @@
+export interface ButtonsComponentSettings {
+    createOrder?: (data: object, actions: object) => void;
```

Comment:
Should be able to enhance this with the Types from https://gist.github.com/camsjams/b843637df7a30c260b341daf97a96d60#file-paypal-react-paypal-js-d-ts-L147